src/components/form-elements/draft-js-mention-selector/DraftJSMentionSelector.js

```diff
@@ -75,27 +75,83 @@ class DraftJSMentionSelector extends React.Component<Props, State> {
         // otherwise we initialize it here
         this.state = {
             contacts: [],
-            hasReceivedFirstInteraction: false,
+            isTouched: false,
             internalEditorState: props.editorState ? null : EditorState.createEmpty(mentionDecorator),
             error: null,
         };
     }
 
-    /**
-     * Lifecycle method that gets called when a component is receiving new props
-     * @param {object} nextProps Props the component is receiving
-     * @returns {void}
-     */
-    componentWillReceiveProps(nextProps: Props) {
+    static getDerivedStateFromProps(nextProps: Props) {
         const { contacts } = nextProps;
+        let nextState = null;
 
         if (contacts) {
-            this.setState({ contacts }, () => {
-                this.checkValidity();
-            });
+            nextState = { contacts };
+        }
+
+        return nextState;
+    }
+
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        const { internalEditorState: prevInternalEditorState } = prevState;
+        const { internalEditorState } = this.state;
+
+        const { editorState: prevEditorStateFromProps } = prevProps;
+        const { editorState } = this.props;
+
+        let prevEditorState;
+        let currentEditorState;
+
+        // Determine whether we're working with the internal editor state or
+        // external editor state passed in from props
+        if (prevInternalEditorState) {
```

Comment:
It looks like `this.state.internalEditorState` is `null` if the component is controlled. Would something like the below work?

```
const prevEditorState = prevInternalEditorState || prevEditorStateFromProps;
const currentEditorState = internalEditorState || editorState;
```